src/Plover/Reduce.hs

```diff
@@ -212,6 +213,13 @@ typeCheck' e@(a :<= b) = do
   tr <- typeCheck' b
   unifyGen tl tr
   return Void
+typeCheck' e@(If cond t f) = do
+  local $ typeCheck' t
+  local $ typeCheck' f
+  ctype <- typeCheck' cond
+  case ctype of
+    BoolType -> return Void
```

Comment:
Whats the solution? Make sure both branches have the same type and then make the if statement also inherit that type?